protocol/v2/ssv/validator/non_committee_validator.go

```diff
@@ -250,6 +257,13 @@ func (ncv *CommitteeObserver) processMessage(
 			}
 		}
 	}
+
+	// Remove older slots container
+	thresholdSlot := currentSlot - 12 // won't get new messages after 12 slots
+	if _, exists := ncv.postConsensusContainer[thresholdSlot]; exists {
+		delete(ncv.postConsensusContainer, thresholdSlot)
```

Comment:
Do I understand correctly that it's possible that if there are no messages in a slot, there is a gap and older messages won't be removed? If so, perhaps it would be great to leave a comment that it's very unlikely to happen, so it shouldn't be an issue

Reply:
yeah this could cause unbounded growth because for every skipped slot X the map will keep slot X-12 forever

i think we could try something like this instead:

```go
const containerSize phase0.Slot = 34
// ...
postConsensusContainer [containerSize]map[phase0.ValidatorIndex]*ssv.PartialSigContainer

// Access:
index := msg.Slot%containerSize
postConsensusContainer[index] = ...
```

Reply:
@nkryuchkov sure, this pattern could be coded to something like:

```go
type SlotContainer[T any] struct {
  capacity int
  items []T
}

Get(phase0.Slot)
Set(phase0.Slot, T)
```

however now that i think of it, isn't there a risk of a early or late messages overwriting the current working set? (~last 34 slots)

Reply:
@moshe-blox good point, it seems the size should include allowance in the past + future. By the way, we already use 32/34 somewhere, so we need to fix it.

Also, I think we need to bind it to message validation somehow because if we change it, it will be hard to find all places where it's used

Reply:
see this https://github.com/ssvlabs/ssv/pull/1859#discussion_r1850799907
of course we could implement something more elegant but for now we want to see the impact of just removing from the map :)

Reply:
it doesn't appear in pprof so IMO we can leave it as is for now, but I left a comment for the future